lib/ivis_opengl/pielighting.cpp

```diff
@@ -0,0 +1,199 @@
+			center + horizontal * range + vertical * range + forward * range
+		};
+	}
+
+}
+
+void renderingNew::LightingManager::ComputeFrameData(const LightingData& data, const glm::mat4& worldViewProjectionMatrix)
+{
+	WZ_PROFILE_SCOPE("LightingManager_ComputeFrameData");
+	PointLightBuckets result;
+
+	// Pick the first lights inside the view frustrum
+	auto viewFrustrum = IntersectionOfHalfSpace{
+		[](glm::vec3 in) { return in.x >= -1.f; },
+		[](glm::vec3 in) { return in.x <= 1.f; },
+		[](glm::vec3 in) {
+			if (gfx_api::context::get().isYAxisInverted())
+				return -in.y >= -1.f ;
+			return in.y >= -1.f ;
+		},
+		[](glm::vec3 in) {
+			if (gfx_api::context::get().isYAxisInverted())
+				return -in.y <= 1.f;
+			return in.y <= 1.f;
+		},
+		[](glm::vec3 in) { return in.z >= 0; },
+		[](glm::vec3 in) { return in.z <= 1; }
+	};
+
+	std::vector<LIGHT> culledLights;
+	for (const auto& light : data.lights)
+	{
+		if (culledLights.size() >= gfx_api::max_lights)
+			break;
+		auto clipSpaceBoundingBox = transformBoundingBox(worldViewProjectionMatrix, getLightBoundingBox(light));
+		if (!isBBoxInClipSpace(viewFrustrum, clipSpaceBoundingBox))
+			continue;
+		culledLights.push_back(light);
+	}
+
+
+	for (int lightIndex = 0; lightIndex < culledLights.size(); lightIndex++)
+	{
+		const auto& light = culledLights[lightIndex];
+		result.positions[lightIndex].x = light.position.x;
+		result.positions[lightIndex].y = light.position.y;
+		result.positions[lightIndex].z = light.position.z;
+		result.colorAndEnergy[lightIndex].x = light.colour.byte.r / 255.f;
+		result.colorAndEnergy[lightIndex].y = light.colour.byte.g / 255.f;
+		result.colorAndEnergy[lightIndex].z = light.colour.byte.b / 255.f;
+		result.colorAndEnergy[lightIndex].w = light.range;
+	}
+
+	// Iterate over all buckets
+	int overallId = 0;
+	int bucketId = 0;
+	for (int i = 0; i < gfx_api::bucket_dimension; i++)
+	{
+		for (int j = 0; j < gfx_api::bucket_dimension; j++)
+		{
+			auto frustrum = IntersectionOfHalfSpace{
+				[&](glm::vec3 in) { return in.x >= -1.f + 2 * static_cast<float>(i) / gfx_api::bucket_dimension; },
+				[&](glm::vec3 in) { return in.x <= -1.f + 2 * static_cast<float>(i + 1) / gfx_api::bucket_dimension; },
+				[&](glm::vec3 in) {
+					if (gfx_api::context::get().isYAxisInverted())
+						return -in.y >= -1.f + 2 * static_cast<float>(j) / gfx_api::bucket_dimension;
+					return in.y >= -1.f + 2 * static_cast<float>(j) / gfx_api::bucket_dimension;
+				},
+				[&](glm::vec3 in) {
+					if (gfx_api::context::get().isYAxisInverted())
+						return -in.y <= -1.f + 2 * static_cast<float>(j + 1) / gfx_api::bucket_dimension;
+					return in.y <= -1.f + 2 * static_cast<float>(j + 1) / gfx_api::bucket_dimension;
+				},
+				[&](glm::vec3 in) { return in.z >= 0; },
+				[&](glm::vec3 in) { return in.z <= 1; }
+			};
+
+			int bucketSize = 0;
+			for (int lightIndex = 0; lightIndex < culledLights.size(); lightIndex++)
```

Comment:
Prefer using `size_t` for container indices when iterating. This can safely represent all possible indices and is resistant to integer overflow errors.